src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/CSharpCodeWriter.cs

```diff
@@ -26,44 +26,183 @@ public class CSharpCodeWriter : CodeWriter
             '\u2029',
         };
 
-        public new CSharpCodeWriter Write(string data)
+        private static readonly char[] NewLineCharacters = { '\r', '\n' };
+
+        private string _cache = string.Empty;
+        private bool _dirty;
+
+        private int _absoluteIndex;
+        private int _currentLineIndex;
+        private int _currentLineCharacterIndex;
+
+        public StringBuilder Builder { get; } = new StringBuilder();
+
+        public int CurrentIndent { get; private set; }
+
+        public bool IsAfterNewLine { get; private set; }
+
+        public string NewLine { get; set; } = Environment.NewLine;
+
+        public CSharpCodeWriter ResetIndent()
         {
-            return (CSharpCodeWriter)base.Write(data);
+            return SetIndent(0);
         }
 
-        public new CSharpCodeWriter Indent(int size)
+        public CSharpCodeWriter IncreaseIndent(int size)
         {
-            return (CSharpCodeWriter)base.Indent(size);
+            CurrentIndent += size;
+
+            return this;
         }
 
-        public new CSharpCodeWriter ResetIndent()
+        public CSharpCodeWriter DecreaseIndent(int size)
         {
-            return (CSharpCodeWriter)base.ResetIndent();
+            CurrentIndent -= size;
+
+            return this;
         }
 
-        public new CSharpCodeWriter SetIndent(int size)
+        public CSharpCodeWriter SetIndent(int size)
         {
-            return (CSharpCodeWriter)base.SetIndent(size);
+            CurrentIndent = size;
+
+            return this;
         }
 
-        public new CSharpCodeWriter IncreaseIndent(int size)
+        public CSharpCodeWriter Indent(int size)
```

Comment:
A few methods like this one are unused outside this class. But I've left them as public it makes sense to have this functionality. Let me know if you feel otherwise.